slackbridge/bots.py

```diff
@@ -229,17 +227,21 @@ def signedOn(self):
         # And if not, register for the first time
         self.msg('NickServ', 'GROUP {} {}'.format(self.target_group_nick,
                                                   self.nickserv_password))
-        for channel in self.channels:
-            self.log(log.msg, 'Joining #{}'.format(channel['name']))
-            self.join_channel(channel['name'])
+        for channel_name in self.joined_channels:
+            self.log(log.msg, 'Joining #{}'.format(channel_name))
+            self.join_channel(channel_name)
 
         self.away('Default away for startup.')
 
     def join_channel(self, channel_name):
         self.join('#{}'.format(channel_name))
+        if channel_name not in self.joined_channels:
+            self.joined_channels.append(channel_name)
 
     def part_channel(self, channel_name):
         self.leave('#{}'.format(channel_name))
+        if channel_name in self.joined_channels:
+            self.joined_channels.remove(channel_name)
```

Comment:
Same thing here. The ```left``` method seems like a better place to remove channels. Then you can safely do
```python
if channel_name in self.joined_channels:
    self.leave('#{}'.format(channel_name)
```
in ```part_channel```